### PR TITLE
fix(extra730): Handle invalid date formats checking ACM certificates

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context 

extra730 checks if ACM Certificates are about to expire in `$DAYS_TO_EXPIRE_THRESHOLD` days or less

### Description

This PR addresses the following:
- Modify `bsd_timestamp_to_date` and `gnu_timestamp_to_date` in order to perform date validation properly.
- Handle timestamp errors when an ACM certificate has not expiration date.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/os_detector [hunk 1]
```diff
@@ -32,28 +32,20 @@ bsd_how_older_from_today() {
 # function to convert from timestamp to date
 # output date format %Y-%m-%d
 gnu_timestamp_to_date() {
-  # if date comes from cli v2 in format like 2020-04-29T10:13:09.191000-04:00
-  # we have to get only '%Y-%m-%d'
-  if [[ $1 = 20* ]];then 
-    echo $1 | cut -f1 -d"T"
-  else
```
Comment (jfagoagas) [inline include/os_detector:39]: I've deleted this date parsing because it allows something like `20wathever`

File: include/os_detector [hunk 2]
```diff
@@ -32,28 +32,20 @@ bsd_how_older_from_today() {
...
+  TIMESTAMP_TO_CONVERT=$(cut -f1 -d"." <<<  "${1}")
+  OUTPUT_DATE=$("${DATE_CMD}" -d @"${TIMESTAMP_TO_CONVERT}" +'%Y-%m-%d')
+  echo "${OUTPUT_DATE}"
 }
 bsd_timestamp_to_date() {
-  # if date comes from cli v2 in format like 2020-04-29T10:13:09.191000-04:00
-  # we have to get only '%Y-%m-%d'
-  if [[ $1 = 20* ]];then 
-    echo $1 | cut -f1 -d"T"
-  else
```
Comment (jfagoagas) [inline include/os_detector:51]: I've deleted this date parsing because it allows something like `20wathever`

File: checks/check_extra730 [hunk 3]
```diff
@@ -27,24 +27,34 @@ CHECK_DOC_extra730='https://docs.aws.amazon.com/config/latest/developerguide/acm
...
 extra730(){
+  # Only RSA key types, needed to recover Amazon Issued, Imported and Private PKIs
+  local ACM_KEY_TYPES="RSA_1024,RSA_2048,RSA_3072,RSA_4096"
+  local ACM_CERTIFICATE_STATUSES="ISSUED"
+
   # "Check if ACM Certificates are about to expire in $DAYS_TO_EXPIRE_THRESHOLD days or less"
   for regx in $REGIONS; do
-    LIST_OF_ACM_CERTS=$($AWSCLI acm list-certificates $PROFILE_OPT --region $regx --query 'CertificateSummaryList[].CertificateArn' --output text)
-    if [[ $LIST_OF_ACM_CERTS ]];then
+    LIST_OF_ACM_CERTS=$("${AWSCLI}" acm list-certificates ${PROFILE_OPT} --region "${regx}" --include keyTypes="${ACM_KEY_TYPES}" --certificate-statuses "${ACM_CERTIFICATE_STATUSES}" --query 'CertificateSummaryList[].CertificateArn' --output text)
```
Comment (toniblyx) [inline checks/check_extra730:36]: What about imported certificates?

**Review by toniblyx — CHANGES_REQUESTED** [review]

File: checks/check_extra730 (hunk 3 above)
Comment (jfagoagas) [reply checks/check_extra730:36]: It checks every ACM certificate whether it is Amazon Issued, Imported or comes from a Private PKIs.

For that reason we have included two filters:
- `ACM_KEY_TYPES`: to set the algorithms that can be used to generate key pairs.
- `ACM_CERTIFICATE_STATUSES`: to only check ACM Certificates whose status is `ISSUED`. The following statuses are not checked:
```
PENDING_VALIDATION
INACTIVE
EXPIRED
VALIDATION_TIMED_OUT
REVOKED
FAILED
```

Comment (jfagoagas) [reply checks/check_extra730:36]: It checks every ACM certificate whether it is Amazon Issued, Imported or comes from a Private PKI.

For that reason we have included two filters:
- `ACM_KEY_TYPES`: to set the algorithms that can be used to generate key pairs.
- `ACM_CERTIFICATE_STATUSES`: to only check ACM Certificates whose status is `ISSUED`. The following statuses are not checked:
```
PENDING_VALIDATION
INACTIVE
EXPIRED
VALIDATION_TIMED_OUT
REVOKED
FAILED
```

File: include/os_detector [hunk 4]
```diff
@@ -32,28 +32,20 @@ bsd_how_older_from_today() {
...
-    TIMESTAMP_TO_CONVERT=$(echo $1 | cut -f1 -d".")
-    OUTPUT_DATE=$("$DATE_CMD" -d @$TIMESTAMP_TO_CONVERT +'%Y-%m-%d')
-    echo $OUTPUT_DATE
-  fi 
+  # if date comes from cli v2 in format like 2020-04-29T10:13:09.191000-04:00, which is ISO8601
+  
+  # remove fractions of a second
+  TIMESTAMP_TO_CONVERT=$(cut -f1 -d"." <<<  "${1}")
+  OUTPUT_DATE=$("${DATE_CMD}" -d @"${TIMESTAMP_TO_CONVERT}" +'%Y-%m-%d')
+  echo "${OUTPUT_DATE}"
```
Comment (n4ch04) [inline include/os_detector:40]: Is this echo really needed?

Comment (jfagoagas) [reply include/os_detector:40]: Yes, it's the way to return the `OUTPUT_DATE` value

**Review by n4ch04 — APPROVED** [review]

**Review by toniblyx — APPROVED** [review]

**Pull request merged**